feat: add Comment type and createComment mutation to GraphQL schema

Farid-Efrei opened the pull request:
This pull request includes several important changes to the GraphQL-Project-Social-Network repository, focusing on adding issue templates, a code of conduct, contributing guidelines, and security policies, as well as enhancing the project setup and configuration.

### Documentation and Community Guidelines:

* **Bug Report Template**: Added a template for reporting bugs to help improve the project. This includes sections for description, steps to reproduce, expected behavior, actual behavior, environment, additional context, and related issues. (.github/ISSUE_TEMPLATE/bug_report.md)
* **Feature Request Template**: Added a template for suggesting new features. This includes sections for description, problem or use case, proposed solution, alternatives considered, additional context, and related issues. (.github/ISSUE_TEMPLATE/feature_request.md)
* **Code of Conduct**: Implemented the Contributor Covenant Code of Conduct to ensure a welcoming and inclusive community. (CODE_OF_CONDUCT.md)
* **Contributing Guidelines**: Detailed guidelines for contributing to the project, including setup instructions, how to report bugs, suggest features, and submit code changes. (CONTRIBUTING.md)
* **Security Policy**: Added a security policy outlining supported versions, how to report vulnerabilities, and the disclosure process. (SECURITY.md)

### Project Setup and Configuration:

* **Pre-commit Hook**: Added a pre-commit hook to run lint-staged checks, ensuring code quality before commits. (.husky/pre-commit)
* **npm Configuration**: Enabled automatic installation of peer dependencies. (.npmrc)
* **Server Configuration**: Updated server configuration files including gitignore for Prisma database files, Biome configuration, and code generation setup. (apps/server/.gitignore, apps/server/biome.json, apps/server/codegen.ts) [[1]](diffhunk://#diff-73a249eca23b4a6165a70cd4d1534c5b9e65e734faad0b578b2288938ae623abR1-R3) [[2]](diffhunk://#diff-1f9cb96a9f943c51ef5f602ff4eb2e695ebc1d56d708ce45d1a856f7ca43c90cR1-R7) [[3]](diffhunk://#diff-6989d66dc4f37000bf38c8dd7a9dbd6edcfc1c2c4159683dc72547b7561a306aR1-R16)
* **Server Package.json**: Added necessary dependencies and scripts for the server application. (apps/server/package.json)

These changes collectively enhance the project's documentation, community engagement, and development workflow.

georgyguei commented on apps/server/src/resolvers.ts:
```diff
@@ -50,6 +57,21 @@ export const resolvers: Resolvers = {
+          ...comment,
+          createdAt: comment.createdAt.toISOString(),
+          updatedAt: comment.updatedAt.toISOString(),
+          // article: {
+          //   ...comment.article,
+          //   createdAt: comment.article.createdAt.toISOString(),
+          //   updatedAt: comment.article.updatedAt.toISOString(),
+          // },
```
Remove useless comments

georgyguei commented on apps/server/src/resolvers.ts:
```diff
@@ -170,6 +240,55 @@ export const resolvers: Resolvers = {
+          authorId: context.user.id,
+        },
+        include: {
+          //  article: true,
```
Remove useless comment

georgyguei commented on apps/server/src/resolvers.ts:
```diff
@@ -170,6 +240,55 @@ export const resolvers: Resolvers = {
+        ...comment,
+        createdAt: comment.createdAt.toISOString(),
+        updatedAt: comment.updatedAt.toISOString(),
+        // article: {
+        //   ...comment.article,
+        //   author: {
+        //     ...comment.author,
+        //     createdAt: comment.author.createdAt.toISOString(),
+        //     updatedAt: comment.author.updatedAt.toISOString(),
+        //   },
+        //   createdAt: comment.article.createdAt.toISOString(),
+        //   updatedAt: comment.article.updatedAt.toISOString(),
+        // },
+
```
Remove useless comments

Farid-Efrei commented:
This pull request introduces significant changes to the GraphQL schema and resolvers in the server application to support comments on articles. It includes modifications to the resolvers, schema definitions, and type definitions to handle comments.

### GraphQL Resolvers Enhancements:
* Added support for including comments and their authors when fetching articles and articles by author. (`apps/server/src/resolvers.ts`, [[1]](diffhunk://#diff-2b1f559e35df3c6aee138dc7111aa00f903383b67bdf8a8f74d7b9e2d7278d06L39-R46) [[2]](diffhunk://#diff-2b1f559e35df3c6aee138dc7111aa00f903383b67bdf8a8f74d7b9e2d7278d06R60-R70) [[3]](diffhunk://#diff-2b1f559e35df3c6aee138dc7111aa00f903383b67bdf8a8f74d7b9e2d7278d06R81-R88) [[4]](diffhunk://#diff-2b1f559e35df3c6aee138dc7111aa00f903383b67bdf8a8f74d7b9e2d7278d06R98-R107) [[5]](diffhunk://#diff-2b1f559e35df3c6aee138dc7111aa00f903383b67bdf8a8f74d7b9e2d7278d06L85-R119) [[6]](diffhunk://#diff-2b1f559e35df3c6aee138dc7111aa00f903383b67bdf8a8f74d7b9e2d7278d06R130-R139) [[7]](diffhunk://#diff-2b1f559e35df3c6aee138dc7111aa00f903383b67bdf8a8f74d7b9e2d7278d06R168) [[8]](diffhunk://#diff-2b1f559e35df3c6aee138dc7111aa00f903383b67bdf8a8f74d7b9e2d7278d06L133-R198) [[9]](diffhunk://#diff-2b1f559e35df3c6aee138dc7111aa00f903383b67bdf8a8f74d7b9e2d7278d06R208-R217) [[10]](diffhunk://#diff-2b1f559e35df3c6aee138dc7111aa00f903383b67bdf8a8f74d7b9e2d7278d06L163-R229)
* Introduced a new mutation `createComment` to allow users to add comments to articles. (`apps/server/src/resolvers.ts`, [apps/server/src/resolvers.tsR239-R276](diffhunk://#diff-2b1f559e35df3c6aee138dc7111aa00f903383b67bdf8a8f74d7b9e2d7278d06R239-R276))

### GraphQL Schema Updates:
* Added `comments` field to the `Article` type and defined a new `Comment` type. (`apps/server/src/schema.ts`, [apps/server/src/schema.tsR15-R41](diffhunk://#diff-ca76cab8bd94b74775374ab503f17e1b9e8d9c276f273ff77cf49ddca5ce3b54R15-R41))
* Added `createComment` mutation to the schema. (`apps/server/src/schema.ts`, [apps/server/src/schema.tsR50](diffhunk://#diff-ca76cab8bd94b74775374ab503f17e1b9e8d9c276f273ff77cf49ddca5ce3b54R50))

### Type Definitions Adjustments:
* Updated type definitions to include `Comment` type and its related fields. (`apps/server/src/types.ts`, [apps/server/src/types.tsL1-R147](diffhunk://#diff-df69a8eee2e8530ab6b2d9eebe27806a821850ecd9eaa9dee7aa1b541f0aa04cL1-R147))
* Refactored existing type definitions for consistency and clarity. (`apps/server/src/types.ts`, [apps/server/src/types.tsL164-R178](diffhunk://#diff-df69a8eee2e8530ab6b2d9eebe27806a821850ecd9eaa9dee7aa1b541f0aa04cL164-R178))

These changes enhance the functionality of the server by allowing users to interact with comments on articles, providing a more comprehensive and interactive experience.

Farid-Efrei commented:
Reopen
